odbc-api/tests/integration.rs

```diff
@@ -2129,3 +2129,33 @@ fn many_diagnostic_messages() {
 
     // We do not have an explicit assertion, we are just happy if no integer addition overflows.
 }
+
+/// Ensures access to driver and data source info is synchronized correctly when multiple threads
+/// attempt to query it at the same time. First, we query the list of the known drivers and data
+/// sources on the main thread. Then we spawn multiple threads that attempt to query these lists in
+/// parallel. Finally we compare the lists to ensure they match the list we found on the main
+/// thread.
+#[test_case(MSSQL; "Microsoft SQL Server")]
+#[test_case(MARIADB; "Maria DB")]
+#[test_case(SQLITE_3; "SQLite 3")]
+#[test]
+fn synchronized_access_to_driver_and_data_source_info(_: &Profile) {
+    let expected_drivers_len = ENV.drivers().unwrap().len();
+    let expected_data_sources_len = ENV.data_sources().unwrap().len();
+
+    // Verify we have at least one driver to test. If no data sources are available we can still
+    // proceed anyway.
+    assert!(expected_drivers_len > 0);
```

Comment:
That is very thoughtful, but it would kind of generate a false positive if someone is developing directly on a host system withouth any drivers. Sure other tests would fail also. Or did you notice some really strange stuff happening if no driver had been available?

Reply:
Currently the integration tests already fail if drivers aren't available, so I thought we would just enforce it here too. I don't have a preference either way though, we could remove it if we'd like the test to pass if no drivers are available

Reply:
Removed this because it seems fine to allow the test pass if none exist